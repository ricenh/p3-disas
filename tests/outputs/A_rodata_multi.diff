--- outputs/A_rodata_multi.txt	2023-11-02 16:25:14.132238248 -0400
+++ expected/A_rodata_multi.txt	2023-10-10 11:54:35.000000000 -0400
@@ -1,11 +1,6 @@
-Usage: ../y86 <option(s)> mini-elf-file
- Options are:
-  -h      Display usage
-  -H      Show the Mini-ELF header
-  -a      Show all with brief memory
-  -f      Show all with full memory
-  -s      Show the program headers
-  -m      Show the memory contents (brief)
-  -M      Show the memory contents (full)
-  -d      Disassemble code contents
-  -D      Disassemble data contents
+Disassembly of data contents:
+  0x200:                               | .pos 0x200 rodata
+  0x200: 66 6f 6f 00                   |   .string "foo"
+  0x204: 77 30 30 74 00                |   .string "w00t"
+  0x209: 75 70 20 64 6f 77 6e 00       |   .string "up down"
+
