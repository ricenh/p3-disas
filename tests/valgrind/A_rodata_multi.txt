==1602294== Memcheck, a memory error detector
==1602294== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==1602294== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==1602294== Command: ../y86 -D inputs/rodata_multi.o
==1602294== 
Disassembly of data contents:
  0x200:                               | .pos 0x200 rodata
  0x200: 66 6f 6f 00                   |   .string "foo"
  0x204: 77 30 30 74 00                |   .string "w00t"
  0x209: 75 70 20 64 6f 77 6e 00       |   .string "up down"

==1602294== 
==1602294== HEAP SUMMARY:
==1602294==     in use at exit: 0 bytes in 0 blocks
==1602294==   total heap usage: 4 allocs, 4 frees, 20,952 bytes allocated
==1602294== 
==1602294== All heap blocks were freed -- no leaks are possible
==1602294== 
==1602294== For lists of detected and suppressed errors, rerun with: -s
==1602294== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
