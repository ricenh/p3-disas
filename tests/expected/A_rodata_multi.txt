Disassembly of data contents:
  0x200:                               | .pos 0x200 rodata
  0x200: 66 6f 6f 00                   |   .string "foo"
  0x204: 77 30 30 74 00                |   .string "w00t"
  0x209: 75 70 20 64 6f 77 6e 00       |   .string "up down"

